torch/distributed/_tensor/api.py

```diff
@@ -261,15 +261,21 @@ def __tensor_flatten__(self):
         protocol to inform how to flatten a DTensor to local tensor
         for PT2 tracing
         """
-        return ["_local_tensor"], (self._spec, self.requires_grad)
+        return ["_local_tensor"], {
```

Comment:
can you say why this is better?  is it bc dynamo handles dicts better than tuple? or bc hardcoding a string is handled better than an index? or the extra info is helpful somehow?

Reply:
Sorry - late replies now that I'm slowly cleaning up this stack to make it landable.

Yeah I think I'm going to revert this dict change. I originally added it so we could be explicit: if a user is accessing a constant attribute of a subclass in their code, they had better explicitly return it in their tensor_flatten.


But I don't love changing the tensor_flatten API yet again. There probably also isn't a huge benefit to doing this: instead, we can just check if the attribute is an inner tensor, and if not assume that it is constant metadata.

Reply:
Update: I can't do this exactly as described, because I now have no way to dinstinguish between attributes on the subclass, and stuff like methods